### PR TITLE
CR for Job #20367 - Enable getPublicScripts, and getRunningScripts

### PR DESCRIPTION
**Pull request by thoys**
Code Review for Job #20367 - Workitem available at https://www.worklist.net/20367

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: af34add63c5322bffd53f5990a9823a664b21e3b available at: https://gist.github.com/da507517bee77ab3041e<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4610.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4610.dmg

File: examples/example/misc/scriptsExample.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            displayPublicScriptFolder(nodes[i].children, "  " + prefix);
+            continue;
+        }
+        print(prefix + nodes[i].name + " {" + nodes[i].url + "}");
+    }
+}
+
+var publicScripts = Scripts.getPublic();
+print("Public Scripts:");
+displayPublicScriptFolder(publicScripts, " - ");
```
Comment (AndrewMeadows) [inline examples/example/misc/scriptsExample.js]: Could we change the name of this file to be something like: **listAllScripts.js** or something else a bit more descriptive?  The **example** part is already known by the path (**.../examples/example/misc/**) so we could remove that, but we could still use a bit more descriptive title.  I don't mind a longer filename if it helps clarify what the script actually does.

File: interface/src/scripting/ScriptsScriptingInterface.h [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+public slots:
+    QVariantList getRunning();
+    QVariantList getPublic();
+    QVariantList getLocal();
+
+private:
+    QVariantList getPublicChildNodes(TreeNodeFolder* parent);
+};
+
+#endif // hifi_ScriptsScriptingInterface_h
```
Comment (AndrewMeadows) [inline interface/src/scripting/ScriptsScriptingInterface.h]: Again,  **ScriptsScriptingInterface** as a class name just doesn't do it for me.  I'd prefer a more descriptive class name but am not sure what it _should_ be.  Maybe:  **ScriptDiscoveryService** ?

It now occurs to me... why would this be its own class?  Why not just add the query methods to whatever class is returned by **Application::getInstance()->getRunningScriptsWidget()** ?

File: examples/example/misc/scriptsExample.js (hunk 1 above)
Comment (thoys) [inline examples/example/misc/scriptsExample.js]: Yeh sure makes sense :)

File: interface/src/scripting/ScriptsScriptingInterface.h (hunk 2 above)
Comment (thoys) [inline interface/src/scripting/ScriptsScriptingInterface.h]: Sure will do that, should I name the JS object ScriptDiscoveryService though?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0cb5ee5bd18889cd808c96b0001e6cd8628cd403 available at: https://gist.github.com/da507517bee77ab3041e<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4610.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4610.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: cf403a7ad8ac0a695377aab94c40bdfbc5e98d47 available at: https://gist.github.com/da507517bee77ab3041e<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4610.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4610.dmg

**Comment by AndrewMeadows** [comment]
This is much better!  I will try to test it.

**Pull request merged**